pkg/controller/node/controller_utils.go

```diff
@@ -304,7 +304,7 @@ func terminatePods(kubeClient clientset.Interface, recorder record.EventRecorder
 		if remaining < 0 {
 			remaining = 0
 			glog.V(2).Infof("Removing pod %v after %s grace period", pod.Name, grace)
-			recordNodeEvent(recorder, nodeName, api.EventTypeNormal, "TerminatingEvictedPod", fmt.Sprintf("Pod %s has exceeded the grace period for deletion after being evicted from Node %q and is being force killed", pod.Name, nodeName))
+			recordNodeEvent(recorder, nodeName, nodeUID, api.EventTypeNormal, "TerminatingEvictedPod", fmt.Sprintf("Pod %s has exceeded the grace period for deletion after being evicted from Node %q and is being force killed", pod.Name, nodeName))
```

Comment:
This would be a test to verify the correct node UID was set, as long as
it's not unreasonably difficult to properly abstract.  If it's too complex
I'm fine with not including it.

On Sun, Jul 24, 2016 at 6:56 AM, TonyAdo <notifications@github.com> wrote:

> In pkg/controller/node/controller_utils.go
> <https://github.com/kubernetes/kubernetes/pull/29437#discussion_r71986570>
> :
>
> > @@ -304,7 +304,7 @@ func terminatePods(kubeClient clientset.Interface, recorder record.EventRecorder
> >  		if remaining < 0 {
> >  			remaining = 0
> >  			glog.V(2).Infof("Removing pod %v after %s grace period", pod.Name, grace)
> > -			recordNodeEvent(recorder, nodeName, api.EventTypeNormal, "TerminatingEvictedPod", fmt.Sprintf("Pod %s has exceeded the grace period for deletion after being evicted from Node %q and is being force killed", pod.Name, nodeName))
> > +			recordNodeEvent(recorder, nodeName, nodeUID, api.EventTypeNormal, "TerminatingEvictedPod", fmt.Sprintf("Pod %s has exceeded the grace period for deletion after being evicted from Node %q and is being force killed", pod.Name, nodeName))
>
> @smarterclayton <https://github.com/smarterclayton> I just noticed that
> we have already add such test cases here
> <https://github.com/kubernetes/kubernetes/blob/master/pkg/client/record/event_test.go#L104>.
> So do we still need to add test case in node controller? If so I will do
> that.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/29437/files/6977eaf67e62bcbaeb4a8b44b5116d5b2c73679b#r71986570>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABG_p1E3w9EVZPfjHAG8S06jUPOW2fPlks5qY0TJgaJpZM4JShB4>
> .
>